rdb/src/main/java/com/instana/dc/rdb/DbDcUtil.java

```diff
@@ -203,7 +211,7 @@ public static <T> List<T> getSimpleListWithSql(Connection connection, String que
             ResultSet rs = executeQuery(connection, queryStr);
             if (rs.isClosed()) {
                 logger.severe("getSimpleObjectWithSql: ResultSet is closed");
-                return null;
```

Comment:
There are several usage of getMetricWithSql method to compare the result with null.

Reply:
Updated the reference usages.

Reply:
An empty collection means the statement is successful but no record satisfies the query. 
A null value indicates an exception, an wrong SQL statement if we do not expose the exception directly.
If we can change the return value to return both empty collection and null value based on different scenarios, it will be better than previous. Another good change is to expose the exception.
My original idea which does not throw out the exception is that there might be occasional network issue and the SQL statement can be recovered.
We should not just return empty collection which is to omit the error. It is very useful to help us to find out system problems.
If we do need a function which does not return null, please create another function in Informix to encapsulate this function.

Reply:
Yes, I got your point and updated the method and logged the exception also. Not throwing the exception as I don't want to stop the monitoring for a single metrics issue. 

>If we do need a function which does not return null, please create another function in Informix to encapsulate this function.

No, we don't need it. We will try to maintain same approach for all RDB DC.